src/main/java/org/testng/xml/XmlWeaver.java

```diff
@@ -0,0 +1,80 @@
+package org.testng.xml;
+
+import org.testng.TestNGException;
+import org.testng.internal.ClassHelper;
+import org.testng.internal.reflect.ReflectionRecipes;
+
+/**
+ * A Utility class that helps represent a {@link XmlSuite} and {@link XmlTest} as String.
+ */
+final class XmlWeaver {
+    private static IWeaveXml instance = null;
+    private static boolean testMode = Boolean.parseBoolean(System.getProperty("testng.testmode"));
```

Comment:
Won't be overridden, so it could be final

Reply:
fixed.